### PR TITLE
Add PostgisGeometryCodec

### PR DESCRIPTION
**Pull request by earlbread**
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/pgjdbc/r2dbc-postgresql/blob/main/.github/CONTRIBUTING.adoc).
- [x] You have created a feature request first to discuss your contribution intent. Please reference the feature request ticket number in the pull request.
- [x] You use the code formatters provided [here](https://github.com/pgjdbc/r2dbc-postgresql/blob/master/intellij-style.xml) and have them applied to your changes. Don't submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.

 

#### Issue description

This supports Postgis Geometry type. #483

### CODE REVIEW

File: src/main/java/io/r2dbc/postgresql/codec/BuiltinDynamicCodecs.java [hunk 1]
```diff
@@ -48,6 +49,8 @@
             switch (this) {
                 case HSTORE:
                     return new HStoreCodec(byteBufAllocator, oid);
+                case POSTGIS_GEOMETRY:
```
Comment (mp911de) [inline src/main/java/io/r2dbc/postgresql/codec/BuiltinDynamicCodecs.java:52]: The code should include a guard to verify that the required classes for `jts-core` are on the class path.

Comment (earlbread) [reply src/main/java/io/r2dbc/postgresql/codec/BuiltinDynamicCodecs.java:66]: Added `isSupported` method and changed to register codec only when it returns `true`.

**Comment by earlbread** [comment]
Since `WkbReader` is not thread-safe, I changed `decode` to create a `WkbReader` instance every time.

**Comment by mp911de** [comment]
Codecs aren't really used concurrently, they are rather used sequentially and each connection maintains its own codec instances. In any case, codecs might be used by different threads and that can introduce visibility issues so it's perfectly fine to maintain individual instances.

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged, polished, and backported now.

**Pull request closed without merging**